examples/client/client/tasks.nim

```diff
@@ -22,139 +22,101 @@ type
-
-    of waku1App:
-      if topicSplit[4] == "proto": # should be `rlp` for real decoding
-        try:
-          let
-            # will first need to RLP decode payload for real decoding
-            protoMsg = protocol.ProtocolMessage.decode(
-              wakuMessage.payload)
-            appMetaMsg = protocol.ApplicationMetadataMessage.decode(
-              protoMsg.public_message)
-            chatMsg = protocol.ChatMessage.decode(appMetaMsg.payload)
-
-          # "placeholder handling" of decoded protobuffer that will be iterated
-          # along with changes to what data we signal from the client to the
-          # TUI and how we display/notify re: that data in the TUI
-          message = chatMsg.text
-          timestamp = chatMsg.timestamp.int64
-          username = generateAlias(
-            "0x" & byteutils.toHex(protoMsg.bundles[0].identity))
-
-          fallback = false
-
-        except CatchableError as e:
-          error "error decoding waku/1 message", contentTopic=topic,
-            error=e.msg, payload=string.fromBytes(wakuMessage.payload)
+  let topicResult = ContentTopic.init(wakuMessage.contentTopic)
+
+  if topicResult.isErr:
+    err(fmt"could not parse content topic: {wakuMessage.contentTopic}")
+
+  else:
+    topic = topicResult.get
+
+    # we know how to properly handle messages for only some content topics:
+    # * `/toy-chat/2/{topic}/proto`
+    # * `/waku/1/{topic}/proto` -- should end with `/rfc26` for real decoding
+
+    if topic.isChat2:
+      let chat2Result = Chat2Message.decode(wakuMessage.payload)
+      if chat2Result.isOk:
+        let chat2Message = chat2Result.get
+        message = string.fromBytes(chat2Message.payload)
+        timestamp = chat2Message.timestamp
+        username = chat2Message.nick
+
+        fallback = false
+
+      else:
+        error "error decoding toy-chat/2 message", contentTopic=topic,
+          error=chat2Result.error,
+          payload=string.fromBytes(wakuMessage.payload)
+
+    elif topic.isWaku1:
+      try:
+        let
+          # will first need to RLP decode payload for real decoding
+          protoMsg = protocol.ProtocolMessage.decode(
+            wakuMessage.payload)
+          appMetaMsg = protocol.ApplicationMetadataMessage.decode(
+            protoMsg.public_message)
+          chatMsg = protocol.ChatMessage.decode(appMetaMsg.payload)
+
+        # "placeholder handling" of decoded protobuffer that will be iterated
+        # along with changes to what data we signal from the client to the
+        # TUI and how we display/notify re: that data in the TUI
+        message = chatMsg.text
+        timestamp = chatMsg.timestamp.int64
+        username = generateAlias(
+          "0x" & byteutils.toHex(protoMsg.bundles[0].identity))
+
+        fallback = false
+
+      except CatchableError as e:
+        error "error decoding waku/1 message", contentTopic=topic,
+          error=e.msg, payload=string.fromBytes(wakuMessage.payload)
```

Comment:
Assuming the `decode` procs point to the `decode` proc in `status/private/protocol/chat2_message.nim`, I don't actually see where this would throw. Have you tried removing this `try/except` and see if it compiles?

Reply:
Without it the compiler complained it could raise an unlisted exception, but I don't recall which one, I'll experiment and report back.

Reply:
I'm assuming that is because of `generateAlias` raising an Exception, which was updated in #269 to return a Result type. If #269 lands first, then we could update this proc so that the `try/except` only wraps the `decode` calls.